### PR TITLE
Setup Create-service.component for reverse pipeline

### PR DESCRIPTION
**Pull request by MichaelChinn**

### CODE REVIEW

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      bucket.</p>
+  </div>
+  <div *ngSwitchCase="'sqs'">
+    <div class="row">
+      <div class="col-lg-12">
+        <div>Stream ARN</div>
+        <div class="jz-relative">
+          <input type="text" disabled class="jz-input arn-table" [value]="sqsStreamString">
+          <input type="text" class="jz-input arn-input" placeholder="Stream Name"
+                 [(ngModel)]="eventExpression.SQSstreamARN">
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: SQSQueueARN?

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    </div>
+    <p class="jz-radio-extension-link">Function would be triggered when an object is added or modified inside the
+      bucket.</p>
+  </div>
+  <div *ngSwitchCase="'sqs'">
+    <div class="row">
+      <div class="col-lg-12">
+        <div>Stream ARN</div>
+        <div class="jz-relative">
+          <input type="text" disabled class="jz-input arn-table" [value]="sqsStreamString">
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: queue

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          <input type="text" class="jz-input arn-input" placeholder="Stream Name"
+                 [(ngModel)]="eventExpression.SQSstreamARN">
+        </div>
+      </div>
+    </div>
+    <p class="jz-radio-extension-link">Function would be triggered when an object is added or modified inside the
+      bucket.</p>
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: bucket?

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  selector: 'create-service-aws-events',
+  templateUrl: './create-service-aws-events.component.html',
+  styleUrls: ['./create-service-aws-events.component.scss']
+})
+export class CreateServiceAwsEventsComponent implements OnInit {
+  @Input() type = 'awsEventsNone';
+  @Output() setAWSEvents = new EventEmitter();
+
+
+  public sqsStreamString: string = "arn:aws:sqs:us-west-2:" + environment.aws.account_number + ":stream/";
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts]: why hardcoding the regions?

File: core/jazz_ui/src/app/create-service/create-service/create-service.component.html [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+      <create-service-row title="Service Name" required
+                          subtitle="Name of your API, Microservice or Website that you want to create.">
+        <div class="jz-relative">
+          <input type="text" class="jz-input"
+                 placeholder="Enter Service Name"
+                 [attr.data-state]="serviceNameMessage"
+                 [(ngModel)]="createServiceForm.serviceName"
+                 (blur)="validateServiceName($event)">
+          <img *ngIf="availabilityLoader" class="jz-input-loader" src="../assets/images/icons/loading.png">
+          <div [ngSwitch]="serviceNameMessage" class="jz-input-message">
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service/create-service.component.html]: do you think this kind of validation can be in its own component. 
this component itself is too big and is doing a lot. DO you think breaking this into smaller reusable components can make it better?

File: core/jazz_ui/src/app/create-service/create-service/create-service.component.ts [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+      this.namespaceMessage = 'error-hyphens';
+    } else {
+      this.namespaceMessage = 'valid';
+    }
+    if (~this.namespaceMessage.indexOf('valid') && ~this.serviceNameMessage.indexOf('valid')) {
+      return this.checkServiceNameAvailability();
+    } else if (this.serviceNameMessage === 'success-valid-available') {
+      this.serviceNameMessage = 'valid-available';
+    }
+
```
Comment (svidhani) [inline core/jazz_ui/src/app/create-service/create-service/create-service.component.ts]: extra lines. 
other places as well

File: core/jazz_ui/src/app/pages/landing/landing.component.ts [hunk 7]
```diff
@@ -122,7 +122,7 @@ export class LandingComponent implements OnInit {
             this.buttonText = "GET STARTED NOW";
         }
          if(this.authenticationservice.isLoggedIn() && this.router.url == "/"){
-            this.router.navigateByUrl('/services');
+            // this.router.navigateByUrl('/services');
```
Comment (svidhani) [inline core/jazz_ui/src/app/pages/landing/landing.component.ts]: remove?

File: core/jazz_ui/src/app/primary-components/radio-checkbox/radio-checkbox.component.scss [hunk 8]
```diff
@@ -0,0 +1,82 @@
+input[type="radio"] {
+  // display: none;
```
Comment (svidhani) [inline core/jazz_ui/src/app/primary-components/radio-checkbox/radio-checkbox.component.scss]: remove?

**Comment by svidhani** [comment]
Some description about this PR will be good

File: core/jazz_ui/src/app/primary-components/radio-checkbox/radio-checkbox.component.scss (hunk 8 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/primary-components/radio-checkbox/radio-checkbox.component.scss]: 👍

File: core/jazz_ui/src/app/pages/landing/landing.component.ts (hunk 7 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/pages/landing/landing.component.ts]: 👍

File: core/jazz_ui/src/app/create-service/create-service/create-service.component.ts (hunk 6 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service/create-service.component.ts]: 👍

File: core/jazz_ui/src/app/create-service/create-service/create-service.component.html (hunk 5 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service/create-service.component.html]: I considered this, but I wanted the flag that determines the message and the message itself to be easy to find and in the same component. I've also already reduced the file by more than half and thought more componenting might be overkill.

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts (hunk 4 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts]: I was using an old copy of the files as reference. Will update.

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html (hunk 3 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: 👍

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html (hunk 2 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: 👍

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html (hunk 1 above)
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.html]: 👍

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    let event = {
+      type: this.type,
+      source: '',
+      action: ''
+    };
+
+    switch (this.type) {
+      case 'dynamodb':
+        event["source"] = "arn:aws:dynamodb:us-west-2:" + environment.aws.account_number + ":table/" + this.eventExpression.dynamoTable;
```
[the remark this answers is not in the document]
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts]: will address

File: core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        event["action"] = "PutItem";
+        break;
+      case 'kinesis':
+        event["source"] = "arn:aws:kinesis:us-west-2:" + environment.aws.account_number + ":stream/" + this.eventExpression.streamARN;
+        event["action"] = "PutRecord";
+        break;
+      case 's3':
+        event["source"] = this.eventExpression.S3BucketName;
+        event["action"] = "S3:" + this.eventExpression.S3BucketName + ":*";
```
[the remark this answers is not in the document]
Comment (MichaelChinn) [reply core/jazz_ui/src/app/create-service/create-service-aws-events/create-service-aws-events.component.ts]: This is what we have been using

**Comment by suryajak** [comment]
@devsatishm / @chimpuz:  Is this PR defunct?

**Comment by suryajak** [comment]
Closing this PR for lack of response. Please open a new PR if you need one.

**Pull request closed without merging**